src/field_theory/finite.lean

```diff
@@ -42,6 +42,72 @@ def field_of_integral_domain [fintype α] [decidable_eq α] [integral_domain α]
   inv_zero := dif_pos rfl,
   ..show integral_domain α, by apply_instance }
 
+section polynomial
+
+variables [fintype α] [integral_domain α]
+
+open finset polynomial
+
+/-- The cardinality of a field is at most n times the cardinality of the image of a degree n
+  polnyomial -/
+lemma card_image_polynomial_eval [decidable_eq α] {p : polynomial α} (hp : 0 < p.degree) :
```

Comment:
Can't you split this into two pieces? (1) For every polynomial `f` with positive degree there are at most `f.nat_degree` solutions to `f(x) = a`. That sounds like a statement that is "trivial" but will probably be used more than once. (2) If `f : X → Y` is a map with `X` finite, then `card X ≤ n * (range f).card` if for every `y : Y` the inverse image `f^-1(y)` has cardinality `≤ n`. That's a statement that also seems like it might be used in other contexts.

Reply:
This is all done.

Reply:
Boom! Nice! :tada: